pulp_service/pulp_service/app/serializers.py

```diff
@@ -31,24 +37,48 @@ class VulnerabilityReportSerializer(ModelSerializer):
     A serializer for the VulnerabilityReport Model.
     """
 
-    id = serializers.UUIDField()
     vulns = serializers.JSONField()
+    pulp_href = IdentityField(view_name="vuln_report-detail")
 
     class Meta:
         model = VulnerabilityReport
-        fields = ["id", "vulns"]
+        fields = ["vulns", "pulp_href"]
 
 
-class ContentScanSerializer(serializers.Serializer):
+class ContentScanSerializer(serializers.Serializer, ValidateFieldsMixin):
     """
     A serializer for package scan.
     """
 
-    repo_version = serializers.CharField()
+    repo_version = RepositoryVersionRelatedField(
+        required=False,
+        allow_null=True,
+        help_text=_("RepositoryVersion HREF with the packages to be checked."),
+    )
+    package_json = serializers.FileField(
+        required=False,
+        allow_null=True,
+        help_text=_("package-lock.json file with the definition of dependencies to be checked."),
+    )
 
-    def validate_repo_version(self, value):
+    def validate(self, data):
+        data = super().validate(data)
+        if bool(data.get("repo_version", None)) == bool(data.get("package_json", None)):
+            raise serializers.ValidationError(
+                _("Exactly one of 'repo_version' or 'package_json' must be specified.")
+            )
+        return data
+
+    def verify_file(self):
+        uploaded_file = self.validated_data["package_json"]
+        temp_file = PulpTemporaryFile.init_and_validate(uploaded_file, None)
         try:
-            repo_version = RepoVersionHrefPrnFilter.get_repository_version(value)
-        except:
-            raise serializers.ValidationError(_("No matching RepositoryVersion instance found."))
-        return repo_version.pk
+            lock_file_content_json = json.load(uploaded_file)
+            validate(instance=lock_file_content_json, schema=NPM_PACKAGE_LOCK_SCHEMA)
+            temp_file.save()
```

Comment:
Nice! Was there any need to `seek` the uploaded_file back to 0 before calling `json.load`? I notice it is no longer here and I know `init_and_validate` reads through the whole file first to calculate digests.